### PR TITLE
Add escapeStrings option and tests

### PR DESCRIPTION
**Pull request by berstend**
Hi there,
I've followed the discussion in #5 and understand the difficulty to escape strings automatically when necessary.

Unfortunately I need to parse a more complex JSON and double escaping strings is not an option as the file is used in other environments as well.

As in my case escaping all strings by default is not an issue I've added such option (by default false) and additional tests.

Would be great if you would be willing to merge and publish this on NPM, so I don't need to maintain my own fork. 😄

Thanks!

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -37,6 +37,37 @@ describe('Import type test', function() {
     expect(result.css.toString()).to.eql(EXPECTATION);
   });
 
+  it('escapes strings when escapeStrings option is set', function() {
```
Comment (pmowrer) [inline test/index.js]: When the word "when" is used in the test description, it's usually a sign that part of it can be moved into a describe as grouping mechanism. It allows for adding multiple tests for the same condition. E.g.:

```javascript
describe('when escapeStrings options is set', function() {
  it('escapes string values', function() {
     ...
  });

  it('does something else', function() {
    ...
  });
});

File: test/index.js [hunk 2]
```diff
@@ -37,6 +37,37 @@ describe('Import type test', function() {
...
+    let result = sass.renderSync({
+      file: './test/fixtures/escape-strings/style.scss',
+      escapeStrings: true,
+      importer: jsonImporter
+    });
+
+    expect(result.css.toString()).to.eql(escapeStringsExpectation);
+  });
+
+  it(`throws on complex JSON when escapeStrings option is not set`, function() {
```
Comment (pmowrer) [inline test/index.js]: I think "complex JSON" isn't specific enough. You could construct complex JSON that doesn't throw in this test. Something like "throws when parsing a string that isn't a valid SASS expression".

File: test/fixtures/escape-strings/style.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
+@import 'variables.json';
+
+/// Map deep get
+/// @author Hugo Giraudel
+/// @access public
+/// @param {Map} $map - Map
+/// @param {Arglist} $keys - Key chain
+/// @return {*} - Desired value
+@function map-deep-get($map, $keys...) {
```
Comment (pmowrer) [inline test/fixtures/escape-strings/style.scss]: I like the real-world test, but all of this SASS logic kind of obscures what's actually being tested. I think we should be able to simplify this a great deal to really get at what it is that we want to test. Review #5 for some simpler examples, like @diddledan's [original example](https://github.com/Updater/node-sass-json-importer/pull/5#issuecomment-149422978) or @safareli's [list of edge cases](https://github.com/Updater/node-sass-json-importer/issues/7#issuecomment-154046606).

**Comment by berstend** [comment]
Great feedback @pmowrer - thanks!

I will update the PR accordingly. :-)

File: src/index.js [hunk 4]
```diff
@@ -50,6 +54,8 @@ export function parseValue(value) {
     return parseList(value);
   } else if (_.isPlainObject(value)) {
     return parseMap(value);
+  } else if (ESCAPE_STRINGS && _.isString(value)) {
```
Comment (berstend) [inline src/index.js]: Ah, while updating my PR I've realized why I've used the variable as intermediate: `this` is undefined in `parseValue`.

Comment (berstend) [inline src/index.js]: Is there a better way than to put e.g. `var options;` in the outside scope and use that in `parseValue`? 😄

Comment (pmowrer) [inline src/index.js]: Great point! I think the option should be passed to `parseValue` to keep that function pure. 

```javascript
export function parseValue(value, { escapeStrings = false }) {
  ...
```

**Comment by pmowrer** [comment]
Closing due to inactivity

**Comment by esr360** [comment]
Willing to reconsider merging if I pickup and address the PR comments?

**Comment by berstend** [comment]
Thanks a lot @pmowrer for your insightful comments!
Unfortunately I'm super swamped with work stuff and can't fix the PR unfortunately.

@esr360 Feel free to take ownership of this feature :-)

**Comment by pmowrer** [comment]
Go right ahead @esr360! Appreciate your thoughts on the matter

**Comment by pmowrer** [comment]
Closing due to inactivity.
